### PR TITLE
solving #2876

### PR DESCRIPTION
**Pull request by hack3r-0m**
#### Summary

Added description for setting on TLS via NGINX which was missing and conveying information regarding TLS conflict.
#### Ticket Link

#2876

### CODE REVIEW

**Comment by hack3r-0m** [comment]
I have signed CLA, can anyone help me why the build is failing?

**Comment by justinegeffen** [comment]
Hi @hack3r-0m! Thank you for your PR. The part of the build that's failing is for a preview server and the reason it's failing is because it isn't compatible with a forked repo. It doesn't affect your PR at all though. :) Once it's been reviewed and approved, it can be merged. :)

File: source/install/config-tls-mattermost.rst [hunk 1]
```diff
@@ -42,3 +42,18 @@ The easiest option is to set up TLS on the Mattermost Server, but if you expect
 
 .. note::
    Password-protected certificates are not supported.
+   
+**Configure TLS on NGINX(proxy)**:
```
Comment (justinegeffen) [inline source/install/config-tls-mattermost.rst:46]:
```suggestion
**Configure TLS on NGINX (proxy)**
```

File: source/install/config-tls-mattermost.rst [hunk 2]
```diff
@@ -42,3 +42,18 @@ The easiest option is to set up TLS on the Mattermost Server, but if you expect
...
+ 
+ 1. To encrpyt the traffic between End-Client and Mattermost server, NGINX will serve as forward proxy. Communication flow will be as follow:
```
Comment (justinegeffen) [inline source/install/config-tls-mattermost.rst:48]:
```suggestion
 1. NGINX acts as a forward proxy to encrypt the traffic between the client and Mattermost server.
```

File: source/install/config-tls-mattermost.rst [hunk 3]
```diff
@@ -42,3 +42,18 @@ The easiest option is to set up TLS on the Mattermost Server, but if you expect
...
+    End-Client -> NGINX (port 443) -> Mattermost server.
```
Comment (justinegeffen) [inline source/install/config-tls-mattermost.rst:49]:
```suggestion
```

File: source/install/config-tls-mattermost.rst [hunk 4]
```diff
@@ -42,3 +42,18 @@ The easiest option is to set up TLS on the Mattermost Server, but if you expect
...
+ 2. Upstream encryption between NGINX to Mattermost server is allowed. (optional, not needed)
```
Comment (justinegeffen) [inline source/install/config-tls-mattermost.rst:50]:
```suggestion
 2. (Optional) Upstream encryption between NGINX to Mattermost server is allowed.
```

File: source/install/config-tls-mattermost.rst [hunk 5]
```diff
@@ -42,3 +42,18 @@ The easiest option is to set up TLS on the Mattermost Server, but if you expect
...
+ 3. Follow `NGINX's guide on setting up SSL Termination for TCP Upstream Servers <https://docs.nginx.com/nginx/admin-guide/security-controls/terminating-ssl-tcp/>`__.
+ 
+ Other helpful resources:
+ 
+  `NGINX's ssl blog <https://www.nginx.com/blog/nginx-ssl/>`__ and
```
Comment (justinegeffen) [inline source/install/config-tls-mattermost.rst:55]:
```suggestion
- `NGINX's SSL blog <https://www.nginx.com/blog/nginx-ssl/>`__
```

File: source/install/config-tls-mattermost.rst [hunk 6]
```diff
@@ -42,3 +42,18 @@ The easiest option is to set up TLS on the Mattermost Server, but if you expect
...
+  `NGINX's ssl guide <https://docs.nginx.com/nginx/admin-guide/security-controls/terminating-ssl-http/>`__
```
Comment (justinegeffen) [inline source/install/config-tls-mattermost.rst:56]:
```suggestion
- `NGINX's SSLguide <https://docs.nginx.com/nginx/admin-guide/security-controls/terminating-ssl-http/>`__
```

File: source/install/config-tls-mattermost.rst [hunk 7]
```diff
@@ -42,3 +42,18 @@ The easiest option is to set up TLS on the Mattermost Server, but if you expect
...
+ 
+.. note::
+  Do not setup TLS on mattermost before before doing so for NGINX. It breaks the connection as the TLS prevented it from successfully communicating with the Mattermost server.
```
Comment (justinegeffen) [inline source/install/config-tls-mattermost.rst:59]:
```suggestion
  Do not set up TLS on Mattermost before before doing so for NGINX. It breaks the connection as the TLS prevents it from successfully communicating with the Mattermost server.
```

File: source/install/config-tls-mattermost.rst (hunk 1 above)
Comment (justinegeffen) [reply source/install/config-tls-mattermost.rst:46]: Are the steps below a procedure or are they guidelines? If they're guidelines, one option is to replace the numbers with bullet points instead. :)

**Comment by justinegeffen** [comment]
Hi @hack3r-0m. I've provided feedback and suggestions. If you're happy with them you can commit them and then I can do a second pass review.

**Comment by hack3r-0m** [comment]
Thanks a lot, @justinegeffen for your help in improving formatting and language. I understand it needs to precise for the end-user.

File: source/install/config-tls-mattermost.rst [hunk 8]
```diff
@@ -42,3 +42,17 @@ The easiest option is to set up TLS on the Mattermost Server, but if you expect
 
 .. note::
    Password-protected certificates are not supported.
+   
+**Configure TLS on NGINX (proxy)**
```
Comment (justinegeffen) [inline source/install/config-tls-mattermost.rst:46]: This section is called "Configure" but the three steps below seem to be a list of functionality available as opposed to instructions. One option is to rename this header to something like "Using TLS on NGINX as a proxy", with a brief introduction (using point one), followed by points two and three being turned into bullet-points. The more context and detail around this, the better. :)

File: source/install/config-tls-mattermost.rst [hunk 9]
```diff
@@ -42,3 +42,17 @@ The easiest option is to set up TLS on the Mattermost Server, but if you expect
...
+ 2. (Optional) Upstream encryption between NGINX to Mattermost server is allowed.
+ 3. Follow `NGINX's guide on setting up SSL Termination for TCP Upstream Servers <https://docs.nginx.com/nginx/admin-guide/security-controls/terminating-ssl-tcp/>`__.
+ 
+ Other helpful resources:
+ 
+- `NGINX's SSL blog <https://www.nginx.com/blog/nginx-ssl/>`__
+- `NGINX's SSLguide <https://docs.nginx.com/nginx/admin-guide/security-controls/terminating-ssl-http/>`__
+ 
+.. note::
+  Do not set up TLS on Mattermost before before doing so for NGINX. It breaks the connection as the TLS prevents it from successfully communicating with the Mattermost server.
```
Comment (justinegeffen) [inline source/install/config-tls-mattermost.rst:58]: This is a really helpful note. I propose moving it to the beginning of the section for awareness.

File: source/install/config-tls-mattermost.rst (hunk 8 above)
Comment (hack3r-0m) [reply source/install/config-tls-mattermost.rst]: done

File: source/install/config-tls-mattermost.rst (hunk 9 above)
Comment (hack3r-0m) [reply source/install/config-tls-mattermost.rst]: done

File: source/install/config-tls-mattermost.rst [hunk 10]
```diff
@@ -42,3 +42,17 @@ The easiest option is to set up TLS on the Mattermost Server, but if you expect
 
 .. note::
    Password-protected certificates are not supported.
+   
+**Using TLS on NGINX (as a proxy)**
+
+.. note::
+  Do not set up TLS on Mattermost before before doing so for NGINX. It breaks the connection as the TLS prevents it from successfully communicating with the Mattermost server.
+ 
+ - NGINX will act as a forward proxy to encrypt the traffic between the client and Mattermost server. Hence after installing SSL certificate,  the incoming traffic will be handled via NGINX on port 443 exposed to internet, proxy to the Mattermost server running on port 80.
```
Comment (justinegeffen) [inline source/install/config-tls-mattermost.rst:51]:
```suggestion
 - NGINX will act as a forward proxy to encrypt the traffic between the client and Mattermost server. After installing the SSL certificate, the incoming traffic will be handled via NGINX on port 443 exposed to the internet, proxy to the Mattermost server running on port 80.
```

**Review by justinegeffen — APPROVED** [review]

**Review by svelle — APPROVED** [review]
LGTM!

**Comment by jfrerich** [comment]
@justinegeffen, can you merge after final approval? Thanks!

**Pull request merged**